internal/executor/executor.go

```diff
@@ -0,0 +1,177 @@
+package executor
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/operantai/secops-chaos/internal/k8s"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+	"k8s.io/utils/pointer"
+)
+
+type RemoteExecutorConfig struct {
+	Name       string
+	Namespace  string
+	Image      string
+	Parameters RemoteExecutor
+}
```

Comment:
Return the channels as part of the config. 

```suggestion
type RemoteExecutorConfig struct {
	Name       string
	Namespace  string
	Image      string
	Parameters RemoteExecutor
	StopCh <-chan struct{}
	ReadyCh <-chan struct{}
	ErrCh <- chan struct{}
}
```